packages/orm/src/relations/hasAsyncProperty.ts

```diff
@@ -23,7 +23,7 @@ export interface LoadedProperty<T extends Entity, V> {
  * with `someProperty.get`.
  */
 export function hasAsyncProperty<T extends Entity, H extends LoadHint<T>, V>(
-  loadHint: H,
+  loadHint: Const<H>,
```

Comment:
Not entirely sure why, but one of the `Loaded` refactorings meant this `loadHint` was losing it's `const`-ness, so the `fn` lambda was being typed as `Loaded<Book, LoadHint<Book>>` which is too generic, instead of the specific const `Loaded<Book, "authors">` or what not.

We already use this `Const` helper to `const`-ize the params in `EntityManager.populate`, but for some reason it wasn't necessary here before... :shrug: